modules/core/src/smithy4s/Lazy.scala

```diff
@@ -25,6 +25,14 @@ final class Lazy[A](make: () => A) {
   }
 
   def map[B](f: A => B): Lazy[B] = new Lazy(() => f(make()))
+
+  override def equals(obj: Any): Boolean = obj match {
+    case that: Lazy[_] => this.value == that.value
+    case _             => false
+  }
+
+  override def hashCode(): Int = value.hashCode()
```

Comment:
**issue:** this is going to be a problem whenever `Lazy` is used with recursive schemas.

just spitballing, maybe we make a new `Lazy` for hints, which _does_ have equality, and keep the old one for recursion cases (schemas), without equality?

Reply:
oh, it just struck me.

What if we simply don't implement these, and instead implement custom equality/hashcode in `StaticBinding` itself?